File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        diff = 0
        mini = prices[0]
        ans = 0

        for item in prices:
            diff = item - mini
            if diff < 0:
                mini = item
            ans = max(ans, diff)

        return ans
    

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.ts
function maxProfit(prices: number[]): number {
    let diff: number = 0
    let mini: number = prices[0]
    let ans: number = 0

    prices.forEach((item) =>{
        diff = item - mini;
        if (diff < 0) mini = item;
        ans = Math.max(ans, diff)
    });

    return ans;
};